crates/assistant/src/assistant_settings.rs

```diff
@@ -163,6 +163,7 @@ pub enum AssistantProvider {
         model: OpenAiModel,
         api_url: String,
         low_speed_timeout_in_seconds: Option<u64>,
+        available_models: Option<Vec<OpenAiModel>>,
```

Comment:
Have you thought about having this be `Vec<OpenAiModel>` instead? Is it important to be able to have the case `Some(vec![])` work, or does that not make sense?